### PR TITLE
fix: Stop smeshing clean up smeshing-proving-opts in config

### LINKED ISSUE #1385: Stop smeshing clean up smeshing-proving-opts in config
### Environment

* Smapp version: latest

### Describe the bug
 per title 
 
trigger an issue with go-spacemesh node 

### Steps to reproduce

1. Start smapp
2. Start smeshing and select nonces and threads 
3. See these values in node-config.json in the APP_FOLDER
4. Stop smeshing
5. See the smeshing-proving-opts in node-config.json empty 

### Expected behavior

smeshing-proving-opts property should not be rewritten after stop smeshing 
### Actual behavior

smeshing-proving-opts property is empty in the node-config.json

### PR DESCRIPTION
**Pull request by maparr**
It closes #1385 

### How to test:
Screenshot 2023-07-21 at 17 36 57

**A:** missing smeshing-proving-opts 
1. open custom config and remove smeshing-proving-opts ( init the smeshing before )
3. sign in 
4. see the modal
5.  update settings 
6. settings should be updated

**B:** update smeshing-proving-opts 
1. start smashing 
2. click edit 
3. edit PoS proving opts
4. see the smeshing update  
5. settings should be updated

**C:** update smeshing-proving-opts  with enabled auto-start
1. enable auto-start
2. restart PC
3. see the opened app and the modal with update proving settings
4. update settings
5. settings should be updated

**D:** update smeshing-proving-opts  with the broken node by default ( for example remove the boot nodes from the config )
1. start the app
2. node is down
3. see  modal with update proving settings
4. update settings
5. settings should be updated 

Screenshot 2023-07-21 at 16 59 02
Screenshot 2023-07-21 at 16 59 13

Screenshot 2023-07-21 at 17 05 25
Screenshot 2023-07-21 at 16 59 25

Screenshot 2023-07-21 at 17 43 48

###  Checklist
need to test and resolve additional cases from the code 
- [ ] wallet only
- [ ] auto-start

### CODE REVIEW

**Comment by monikasmolarek** [comment]
Changing the `pos proving opts `while the POS generation is paused does not save the update in the configs, even though the change appears on the smeshing screen. Even app restart doesn’t update the configs. Is it meant to be updated only while the smeshing process keeps running/is resumed?

https://github.com/spacemeshos/smapp/assets/89876259/af36a69b-c6a9-4bd5-b9b6-17e22ff7d4ef

File: app/screens/modal/PoSProvingOptsUpdateWarningModal.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    dispatch(omitAllWarningsByType(WarningType.UpdateSmeshingProvingOpts));
+  };
+
+  return (
+    <ReactPortal modalId="pos-profiler-modal">
+      <Modal
+        header={'Set PoS proving opts'}
+        subHeader={
+          'Need to update PoS proving opts settings. They are missing in your config'
```
Comment (monikasmolarek) [inline app/screens/modal/PoSProvingOptsUpdateWarningModal.tsx]:
```suggestion
          'Please update the PoS proving opts. They are missing in your config.'
```

File: app/screens/modal/PoSProvingOptsUpdateModal.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  const smesherConfig = useSelector((state: RootState) => state.smesher.config);
+  const singleCommitmentSize =
+    (smesherConfig.bitsPerLabel * smesherConfig.labelsPerUnit) / BITS;
+
+  return (
+    <ReactPortal modalId="pos-profiler-modal">
+      <Modal
+        header={'Update PoS proving opts'}
+        subHeader={
+          "Allow to update PoS proving opts which include nonce's and threads"
```
Comment (monikasmolarek) [inline app/screens/modal/PoSProvingOptsUpdateModal.tsx]:
```suggestion
          `This screen allows updating the number of Nonces and CPU threads. 
          These values will be used in the proving process.`
```

**Comment by maparr** [comment]
> Changing the `pos proving opts `while the POS generation is paused does not save the update in the configs, even though the change appears on the smeshing screen. Even app restart doesn’t update the configs. Is it meant to be updated only while the smeshing process keeps running/is resumed?
> 
>  Screen.Recording.2023-07-21.at.22.22.mov

after the update, Im writhing to node config, and after it do  stop / start for smeshing process. The Idea was to reuse our code, because we didn't have functionality to update config from client side

**Comment by monikasmolarek** [comment]
There was an announcement asking the users to update manually their proving opts. 
In case someone chose incorrect values - do we want to handle the check and display the modal or leave the users make their mistakes? 
Screenshot 2023-07-22 at 14 42 59

**Comment by monikasmolarek** [comment]
Also, I think there's no question about this case, the modal should pop up, now it doesn;t show up: 

one of the proving opts set to 0 
Screenshot 2023-07-22 at 14 45 10
or one of the proving opts missing: 
Screenshot 2023-07-22 at 14 53 36

**Comment by monikasmolarek** [comment]
@maparr will this one be amended in terms of the smashing opts set to 0 (as discussed on Saturday evening)?

**Pull request closed without merging**